### PR TITLE
15.0 merged mhch

### PR DESCRIPTION
**Pull request by mhch-odoo**
### Description

*Please include a summary of the change, the project task link and/or the ID.*

Link to task: [#ID](https://www.odoo.com/web#model=project.task&id=)

### All Submissions:

* [ ] My commit respects the [Odoo commit guideline](https://www.odoo.com/documentation/15.0/developer/misc/other/guidelines.html#git)
* [ ] My commit message respects the [commit template](https://github.com/odoo-ps/psbe-process/wiki/Commits-message-guidelines#template)
* [ ] I have used pre-commit
* [ ] The PR contains **only** my modification and **no other external** commit

### Sh/Runbot:

* [ ] The commits pass test and the branch is green
* [ ] Unit tests have been implemented / standard ones rewritten
* [ ] The Staging is ISO-Prod and will contain only this dev

### Upgrade:

* [ ] The data affected (*if any*) by the changes has been migrated 

### Maintenance:

* [ ] The versions of the modules affected by the changes have been bumped in their manifests

### CODE REVIEW

File: sakaya_dropship/static/src/js/checkout.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            })
+
+            return this._super.apply(this, arguments);
+        },
+        /**
+         * @override
+         * @private
+         * @param {Object} result
+         */
+        _handleCarrierUpdateResultBadge: function (result) {
```
Comment (fja-odoo) [inline sakaya_dropship/static/src/js/checkout.js:49]: there is an override of this in `website_sale_coupon_delivery`

**Review by fja-odoo — CHANGES_REQUESTED** [review]
We need to make sure that it will not send anything to fedex in the staging :)
This totally breaks other carriers flows are we sure they will not use those in the future and just stick to fedex in drop shipping ?

File: sakaya_dropship/migrations/15.0.1.0.0/pre-10.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from odoo.upgrade import util
```
[the remark this answers is not in the document]
Comment (mhch-odoo) [reply sakaya_dropship/migrations/15.0.1.0.0/pre-10.py:1]: No it's not

File: sakaya_products/models/regions.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from odoo import models, fields
+
+
+class Region(models.Model):
+    _name = "region"
+
+    name = fields.Char("Name")
```
[the remark this answers is not in the document]
Comment (mhch-odoo) [reply sakaya_products/models/regions.py:7]: I'm not sure, I followed the structure they setup with studio

File: sakaya_products/models/product_brand.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+class ProductBrand(models.Model):
+    _name = "product.brand"
+    _description = "Product Brand"
+    _order = "name"
+
+    name = fields.Char("Brand Name", required=True)
+    description = fields.Text(translate=True)
+    partner_id = fields.Many2one("res.partner", string="Partner", help="Select a partner for this brand if any.", ondelete="restrict")
+    logo = fields.Binary("Logo File")
```
[the remark this answers is not in the document]
Comment (mhch-odoo) [reply sakaya_products/models/product_brand.py:12]: Okay

File: sakaya_dropship/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
+from odoo import http, _
+from odoo.http import request
+from odoo.addons.website_sale_delivery.controllers.main import WebsiteSaleDelivery
+from odoo.exceptions import UserError
+
+
+class SakayaWebsiteSaleDelivery(WebsiteSaleDelivery):
+
+    @http.route()
+    def update_eshop_carrier(self, **post):
```
[the remark this answers is not in the document]
Comment (mhch-odoo) [reply sakaya_dropship/controllers/main.py:10]: Had to patch things in the middle of the method

File: sakaya_dropship/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+            res['is_free_delivery'] = not bool(rate['price'])
+            res['error_message'] = rate['warning_message']
+        else:
+            res['status'] = False
+            res['new_carrier_amount_delivery'] = Monetary.value_to_html(0.0, {'display_currency': order.currency_id})
+            res['error_message'] = rate['error_message']
+        return res
+
+    # Patches website_sale_delivery
+    def _get_shop_payment_values(self, order, **kwargs):
```
[the remark this answers is not in the document]
Comment (mhch-odoo) [reply sakaya_dropship/controllers/main.py:61]: What do you mean?

File: sakaya_dropship/models/delivery_carrier.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    _inherit = 'delivery.carrier'
+
+    delivery_type = fields.Selection(selection_add=[('fixed_fifty', '50 Per Bottle'),], ondelete={'fixed_fifty': 'set default'})
+    support_dropship = fields.Boolean("Supports Dropship")
+
+    def available_carriers(self, partner):
+        merchant = self.env.context.get('merchant_id') or self.env['res.partner']
+        merchant = merchant.sudo()
+        res = super().available_carriers(partner)
+        res = res.filtered(lambda x: x.support_dropship)
```
[the remark this answers is not in the document]
Comment (mhch-odoo) [reply sakaya_dropship/models/delivery_carrier.py:20]: That's what they told me. They only do dropshipping. Supporting standard would require further development

**Pull request merged**